### ISSUE TITLE
findOverlaps with type="equal" and a GRangesList

### ISSUE THREAD

**Issue by kasperdanielhansen** [Bioconductor/GenomicRanges#16]
When I use findOverlaps with type="equal" and a GRangesList I get an error:
```
> findOverlaps(gr, grIntrons.24, type = "equal")
Error in match.arg(type) : 
  'arg' should be one of “any”, “start”, “end”, “within”
```
(in this case `gr` is a GRanges and `grIntrons.24` is a GRangesList). 

This is made even more confusing by the generic
```
> findOverlaps
standardGeneric for "findOverlaps" defined from package "IRanges"

function (query, subject, maxgap = -1L, minoverlap = 0L, type = c("any", 
    "start", "end", "within", "equal"), select = c("all", "first", 
    "last", "arbitrary"), ...) 
standardGeneric("findOverlaps")
<bytecode: 0x7f8a47c5aea8>
<environment: 0x7f8a48e3a1e0>
Methods may be defined for arguments: query, subject
Use  showMethods("findOverlaps")  for currently available ones.
```
which strongly suggests `type="equal"` is valid.

**Comment by lawremi** [comment]
I've recently encountered a use case for the "equal" type. I've defined it along the lines of `setequal()`, so duplicates and order are ignored when determining whether two compound ranges are "equal". This is consistent with type "within" which checks for whether one is a subset of the other. If this sounds OK then I will push it.

**Comment by lawremi** [comment]
I guess what I meant is that `type="within"` requires a within-match for all query ranges (so the query is a subset of the subject in a more general sense), while `type="equal"` requires equality for all query ranges, _and_ all subject ranges (so it is more like set equality). This is at the range level, not position level.

**Comment by lawremi** [comment]
That's right. It was easier to program and more efficient to ignore the duplicates and order and was good enough for my use case. That was the only rationale.

**Comment by hpages** [comment]
Looks like @lawremi  pushed this back in March (commit f25a45f33b5df85b2c63b3d21189982e0504a5cb). Any chance Michael you can add a few unit tests and maybe an example in the man page for this new feature?

Feel free to close the issue. Thanks!
